api/src/test/java/com/messagebird/MessageBirdClientTest.java

```diff
@@ -769,4 +772,144 @@ public void testDeleteWebhook() throws NotFoundException, GeneralException, Unau
         messageBirdClientMock.deleteWebhook("id");
         verify(messageBirdServiceMock, times(1)).deleteByID(VOICE_CALLS_BASE_URL + WEBHOOKS, "id");
     }
+
+    @Test
+    public void testListNumbersForPurchase() throws IllegalArgumentException, GeneralException, UnauthorizedException, NotFoundException {
+        final String url = String.format("%s/v1/available-phone-numbers", NUMBERS_CALLS_BASE_URL);
+        final PhoneNumbersResponse mockedResponse = new PhoneNumbersResponse();
```

Comment:
Actually you can add setter in your objects and here set values in mockedResponse. and then assert the values. 
For example: like in testViewTranscription() method. we create mock objects in TestUtil class. In unit test we use this object and verify all fields in it. This way we can verify files filed types etc. Is there any reason not to check all fields in an object ?

Reply:
I didn't see any use case for setters on this class. Adding public setters just so we can test the class in a way it won't be used only complicates the classes public api, especially since it will be used by the client.
I also think a unit test, testing the serialisation of the object from JSON, would be a better test. From my understanding of the code, the above tests simply test the HTTP layer, making sure that the `requestByID` method gets called with valid parameters?
Would it be possible to mock the HTTP response instead in these tests? And assert the HTTP request (URL/Body/Headers)? I can look into that, but might be part of a bigger refactor